### DISCUSSION TITLE
一体化如何创建入口文件, 而不是通过 hooks 启动项目

### DISCUSSION THREAD

**Discussion by achen-11** [midwayjs/midway#3470] [Q&A] [answered]
### Describe the problem（描述问题）

使用一体化创建的项目, 在服务器部署时想要使用pm2 进行多线程管理, 但打包后的项目并没有找到文档中描述的 bootstrap 入口文件, 只有 hooks build 启动的方式? 请问是否有一体化生成入口文件的方法? 或者使用 pm2 管理一体化项目的方法?

### Midway Versions（Midway 版本）

Need to install the following packages:
  midway-version@latest
Ok to proceed? (y) y
>> Start to check your midway component version...

✓ @midwayjs/faas-typings(not installed)
✓ @midwayjs/fc-starter(not installed)
✓ @midwayjs/serverless-http-parser(not installed)
✓ @midwayjs/async-hooks-context-manager(3.12.3)
✓ @midwayjs/axios(not installed)
✓ @midwayjs/bootstrap(3.12.3)
✓ @midwayjs/bull(not installed)
✓ @midwayjs/bull-board(not installed)
✓ @midwayjs/cache(not installed)
✓ @midwayjs/captcha(not installed)
✓ @midwayjs/casbin(not installed)
✓ @midwayjs/casbin-redis-adapter(not installed)
✓ @midwayjs/casbin-typeorm-adapter(not installed)
✓ @midwayjs/code-dye(not installed)
✓ @midwayjs/consul(not installed)
✓ @midwayjs/core(3.12.3)
✓ @midwayjs/cos(not installed)
✓ @midwayjs/cron(not installed)
✓ @midwayjs/cross-domain(not installed)
✓ @midwayjs/decorator(not installed)
✓ @midwayjs/etcd(not installed)
✓ @midwayjs/express-session(not installed)
✓ @midwayjs/faas(not installed)
✓ @midwayjs/grpc(not installed)
✓ @midwayjs/http-proxy(not installed)
✓ @midwayjs/i18n(not installed)
✓ @midwayjs/info(not installed)
✓ @midwayjs/jwt(3.12.3)
✓ @midwayjs/kafka(not installed)
✓ @midwayjs/leoric(not installed)
✓ @midwayjs/mikro(not installed)
✓ @midwayjs/mock(3.12.3)
✓ @midwayjs/mongoose(not installed)
✓ @midwayjs/oss(not installed)
✓ @midwayjs/otel(not installed)
✓ @midwayjs/passport(not installed)
✓ @midwayjs/process-agent(not installed)
✓ @midwayjs/prometheus(not installed)
✓ @midwayjs/prometheus-socket-io(not installed)
✓ @midwayjs/rabbitmq(not installed)
✓ @midwayjs/redis(3.12.3)
✓ @midwayjs/security(not installed)
✓ @midwayjs/sequelize(not installed)
✓ @midwayjs/session(3.12.3)
✓ @midwayjs/socketio(not installed)
✓ @midwayjs/static-file(not installed)
✓ @midwayjs/swagger(not installed)
✓ @midwayjs/tablestore(not installed)
✓ @midwayjs/tags(not installed)
✓ @midwayjs/typegoose(not installed)
✓ @midwayjs/typeorm(not installed)
✓ @midwayjs/upload(not installed)
✓ @midwayjs/validate(not installed)
✓ @midwayjs/version(not installed)
✓ @midwayjs/view(not installed)
✓ @midwayjs/view-ejs(not installed)
✓ @midwayjs/view-nunjucks(not installed)
✓ @midwayjs/web(not installed)
✓ @midwayjs/express(not installed)
✓ @midwayjs/koa(3.12.3)
✓ @midwayjs/ws(not installed)
**********************************************************************
>> Check complete, all versions are healthy.
**********************************************************************

**Comment by czy88840616** [comment, marked as answer]
经过测试，可行的方案如下：

> 为了不污染源码目录，以下方式最好在构建服务上执行

* 1、执行 npm run build，产生 dist 目录
* 2、拷贝 dist/midway.config.js 到根目录
* 3、重命名或者删除根目录的 midway.config.ts 文件
* 4、在根目录创建 bootstrap.js 文件，内容如下
* 5、在根目录执行 NODE_ENV=production bootstrap.js 启动即可


`bootstrap.js` 文件内容如下。

```js
const { Bootstrap } = require('@midwayjs/bootstrap');
const { join } = require('path');
Bootstrap
  .configure({
    appDir: __dirname,
    baseDir: join(__dirname, 'dist'),
    ignore: [
      '**/_client/**',
    ],
    globalConfig: {
      koa: {
        port: 7001,
      }
    }
  })
  .run();
```
